### ISSUE TITLE
Logical infix operator for possibly NULL objects

### ISSUE THREAD

**Issue by wlandau** [HenrikBengtsson/Wishlist-for-R#120]
Some packages define an internal `%||%` operator to make a decision based on whether an object is `NULL`. I find this helps make code more compact and readable.

```r
`%||%` <- function(x, y) {
  if (is.null(x)) {
    y
  } else {
    x
  }
}
```

I personally also use `%|||%` and `%||NA`, but maybe that's a bit much.

```r
`%|||%` <- function(x, y) {
  if (!length(x)) {
    y
  } else {
    x
  }
}

`%||NA%` <- function(x, y) {
  if (anyNA(x)) {
    y
  } else {
    x
  }
}
```

**Comment by gmbecker** [comment]
depending on the purpose, though, %||% could also reasonably be defined as
what you're calling %|||%.

Also, and more importantly, there are some gotchas here. Big, sharp-edged
ones, regarding operator precedence:

> lobstr::ast(x=a %||% b & c)

█─*`&`*

├─█─*`%||%`*

│ ├─*a*

│ └─*b*

└─*c*

**Comment by HenrikBengtsson** [comment]
I think you've been heard;

```r
$ R --vanilla
R Under development (unstable) (2023-10-26 r85413) -- "Unsuffered Consequences"
...
> x <- NULL
> x %||% 42
[1] 42
```

Source: https://github.com/wch/r-source/commit/10df7eac991c297271e16c7a1888240196d540f2

**Comment by HenrikBengtsson** [comment]
Closing; `%||%` is available in R (>= 4.4.0).
